src/java/org/apache/cassandra/tools/CompactionLogAnalyzer.java

```diff
@@ -0,0 +1,573 @@
+                    if (sstablesIndex < 0)
+                        sstablesIndex = indexMap.get("Tot. SSTables");
+                    compactingSstablesIndex = indexMap.getOrDefault("Comp. Sstables", -1);
+                    if (compactingSstablesIndex < 0)
+                        compactingSstablesIndex = indexMap.get("Comp. SSTables");
+                    sizeIndex = indexMap.getOrDefault("Size (bytes)", -1);
+                    if (sizeIndex < 0)
+                        sizeIndex = indexMap.get("Tot. size (bytes)");
+                    compactionsIndex = indexMap.get("Compactions");
+                    readPerSecIndex = indexMap.get("Read (bytes/sec)");
+                    writePerSecIndex = indexMap.get("Write (bytes/sec)");
+                    sizesIndex = indexMap.getOrDefault("Tot/Read/Written", -1);
+                    if (sizesIndex < 0)
+                        sizesIndex = indexMap.get("Tot. comp. size/Read/Written (bytes)");
+                    Tindex = indexMap.get("T");
+                }
+            }
+    }
+
+    static DataPoint parse(String shardId, String dataLine, DataPoint previous) throws java.text.ParseException
+    {
+        String[] data = dataLine.split(",");
+
+        DataPoint dp = new DataPoint();
+        dp.shardId = shardId;
+        dp.timestamp = getTimestamp(data[timestampIndex], previous);
+        dp.bucket = Integer.parseInt(data[bucketIndex]);
+        dp.sstables = Integer.parseInt(data[sstablesIndex]);
+        dp.size = parseHumanReadable(data[sizeIndex]);
+        final String[] compactions = data[compactionsIndex].split("/");
+        dp.compactionsInProgress = Integer.parseInt(compactions[1]);
+        dp.compactionsPending = Integer.parseInt(compactions[0]);
+        dp.readBytesPerSecond = parseHumanReadable(data[readPerSecIndex]);
+        dp.writeBytesPerSecond = parseHumanReadable(data[writePerSecIndex]);
+        String[] sizes = data[sizesIndex].split("/");
+        dp.totalBytes = parseHumanReadable(sizes[0]);
+        dp.remainingReadBytes = dp.totalBytes - parseHumanReadable(sizes[1]);
+        int T = Integer.parseInt(data[Tindex]);
+        int compactingSSTables = Integer.parseInt(data[compactingSstablesIndex].split("/")[1]);
+        int nonCompacting = dp.sstables - compactingSSTables;
+        dp.bucketsAboveT = nonCompacting > T ? 1 : 0;
+        dp.bucketsAboveT2 = nonCompacting > T*T ? 1 : 0;
+        return dp;
+    }
+
+    private static long getTimestamp(String datum, DataPoint previous) throws java.text.ParseException
+    {
+        try
+        {
+            Date date = new SimpleDateFormat(fullDateFormatter).parse(datum);
+            long time = date.getTime();
+            return time;
+        }
+        catch (java.text.ParseException e)
+        {
+            // The older format had short timestamps without day. Make sure we correct these to not go back in time.
+            Date date = new SimpleDateFormat(dateFormatter).parse(datum);
```

Comment:
As well as the parsing of timestamps without day.

Reply:
Fixed in f7c74f02e7a049b72d1f150cca5dd46f437d1b30.